ui/src/tool-panel/alignment-plan-section-infobox-content.tsx

```diff
@@ -115,16 +115,38 @@ const TrackMeterRange: React.FC<TrackMeterRangeProps> = ({ start, end }) => {
     );
 };
 
-export const AlignmentPlanSectionInfoboxContent: React.FC<
-    AlignmentPlanSectionInfoboxContentProps
-> = ({ sections, onHighlightSection }) => {
+const PlanVisibilityToggle: React.FC<{
+    section: AlignmentPlanSection;
+    isVisible: boolean;
+    togglePlanVisibility: (
+        planId: GeometryPlanId,
+        alignmentId: GeometryAlignmentId | undefined,
+    ) => void;
+}> = ({ section, isVisible, togglePlanVisibility }) => {
+    const planId = section.planId;
+
+    return (
+        <div
+            className={styles['alignment-plan-section-infobox__navigation-plan-visibility-toggle']}>
+            {planId && section.isLinked && (
+                <Eye
+                    visibility={isVisible}
+                    onVisibilityToggle={() => {
+                        togglePlanVisibility(planId, section.alignmentId);
+                    }}
+                />
+            )}
+        </div>
+    );
+};
```

Comment:
Tämä on hyvä muutos ja tällaista refaktorointia kannattaisi IMO harjoittaa muuallakin kun tulee vastaan. Vaikka tämä ei aiheuttaisikaan jossain tilanteissa ongelmia, niin ainakin omasta mielestäni on paljon selkeämpää että React-komponentteja ei ole määritelty sisäkkäin